packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -172,8 +172,12 @@ function getSize (variable) {
       return ${guardAgainstPropertyAccessSideEffects('val', '"length"')}
     } else if (${isInstanceOfCoreType('Set', 'val')} || ${isInstanceOfCoreType('Map', 'val')}) {
       return ${guardAgainstPropertyAccessSideEffects('val', '"size"')}
+    } else if (${isInstanceOfCoreType('WeakSet', 'val')} || ${isInstanceOfCoreType('WeakMap', 'val')}) {
+      throw new TypeError('Cannot get size of WeakSet or WeakMap')
+    } else if (typeof val === 'object' && val !== null) {
+      return Object.keys(val).length
```

Comment:
There is no side effects for this one right?

Reply:
Nope, all should be safe here. But thanks for checking 👍